### PR TITLE
Add apply link

### PR DESCRIPTION
**Pull request by David-Sharpe**
Adds the call to the ApplyLink endpoint.  This endpoint is different in that it does not return JSON or XML data.  It will return a redirect link

### CODE REVIEW

File: Source/CBApi/Framework/requests/ApplyLinkRequest.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            get { return "/v2/application/applylink"; }
+        }
+
+        public string Retrieve() 
+        {
+            _request.AddParameter("JobDID", args["JobDID"]);
+            base.BeforeRequest();
+            IRestResponse response = _client.Execute(_request);
+            _AfterRequestEvent(new RequestEventData(_client, _request, response));
+            //CheckForErrors(response);
```
Comment (tkling) [inline Source/CBApi/Framework/requests/ApplyLinkRequest.cs]: Intentionally left commented out?

File: Source/CBApi/Models/ApplyLink.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        public string SiteID { get; set; }
+        public string HostSite { get; set; }
+        public string JApply { get; set; }
+        public string ApplicationEmail { get; set; }
+        public string TrackingID { get; set; }
+        public string Cobrand { get; set; }
+
+        public ApplyLink(NameValueCollection args) {
+            foreach (string arg in args.Keys) {
+                var x = this.GetType().GetProperties();
```
Comment (tkling) [inline Source/CBApi/Models/ApplyLink.cs]: `var x` is set but never used

File: Source/CBApi/CBApi.cs [hunk 3]
```diff
@@ -147,6 +148,16 @@ public class CbApi : ICBApi {
         }
 
         /// <summary>
+        /// Gets the ApplyLink for a Job
+        /// </summary>
+        /// <param name="request">A named value collection of params to pass to the api</param>
+        /// <returns></returns>
```
Comment (tkling) [inline Source/CBApi/CBApi.cs]: I feel like this doc comment should be filled out if we're gonna have doc comments in this project 👯

File: Source/CBApi/Framework/requests/ApplyLinkRequest.cs (hunk 1 above)
Comment (David-Sharpe) [inline Source/CBApi/Framework/requests/ApplyLinkRequest.cs]: Good catch will be removing the line.

File: Source/CBApi.Tests/CBApi/Models/ApplyLinkTest.cs [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        {
+            args = new NameValueCollection() {
+            {"JobDID", "Fake_DID"},
+            {"SiteID", "Fake_Site_ID"},
+            {"HostSite", "US"},
+            {"JApply", "Fake_JApply"},
+            {"ApplicationEmail", "bob@test.org"},
+            {"TrackingID", "Fake_TrackingID"},
+            {"Cobrand", "Fake_Cobrand"}
+            };
```
Comment (tkling) [inline Source/CBApi.Tests/CBApi/Models/ApplyLinkTest.cs]: Style question, not sure if important or not: indention for the collection contents?

File: Source/CBApi/Models/ApplyLink.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        public string ApplicationEmail { get; set; }
+        public string TrackingID { get; set; }
+        public string Cobrand { get; set; }
+
+        public ApplyLink() : this(new NameValueCollection()) { }
+
+        public ApplyLink(NameValueCollection args) {
+            foreach (string arg in args.Keys) {
+                this.GetType().GetProperty(arg).SetValue(this, args[arg], null);
+	        }
```
Comment (tkling) [inline Source/CBApi/Models/ApplyLink.cs]: Style: this brace is to the right a couple spaces.

**Comment by tkling** [comment]
This PR looks good to me! The only 2 comments I wrote have to do with style and my heart will not be broken if this PR merges with no change :thumbsup:

**Pull request merged**